ts/query.go

```diff
@@ -369,12 +369,12 @@ func (is unionIterator) sumAvg() float64 {
 // returned represent the average value within a sample period. Each datapoint's
 // timestamp falls in the middle of the sample period it represents.
 //
-// If data for a time series was collected from multiple sources, datapoints
-// will represent the sum across all sources at the same sample period. The
-// response will contain a list of all sources which were summed to produce the
-// result.
+// If data for the named time series was collected from multiple sources, each
+// returned datapoint will represent the sum of datapoints from all sources at
+// the same time. The returned string slices contains a list of all sources for
+// the metric which were aggregated to produce the result.
 func (db *DB) Query(name string, r Resolution, startNanos, endNanos int64) (
-	*proto.TimeSeriesQueryResult, error) {
+	[]*proto.TimeSeriesDatapoint, []string, error) {
```

Comment:
I believe I should push the Resolution type into the protos package, which would let me pass a proto.TimeSeriesQueryRequest to this method. However, i've decided to do that in another PR, because this one already has a bit of refactoring.